src/CommonProviderImplementation/Helpers.fs

```diff
@@ -171,9 +172,10 @@ module internal ProviderHelpers =
         using (logTime "Loading" sampleOrSampleUri) <| fun _ ->
     
         let tryGetResource() = 
```

Comment:
You are applying the lazy to the output of the function. There's only going to ever be one thread accessing that. Please revert